### PR TITLE
[FEATURE] (Content) - Wave 8 mini expansions

### PR DESCRIPTION
**Pull request by deusabianca**
This consists of The Caverns, The Abyss, Tales of Old Gravehold and promos

### CODE REVIEW

File: src/aer-data/src/ENG/theCaverns/cards.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	developCost: 4,
+    effect: `
+      <p>
+      <b>Attach</b> this to any player's opened breach.<br/>
+	  When a spell is cast from this breach, do
+	  not discard that spell. You cannot cast
+	  that spell again this casting phase.
+      </p>
+      `,
+    keywords: ['develop'],
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/cards.ts:35]: Add 'attach' keyword.

File: src/aer-data/src/ENG/theCaverns/cards.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	developCost: 5,
+    effect: `
+      <p>
+      <b>Attach</b> this to any player's breach.<br/>
+	  When a spell is cast from this breach,
+	  place an elemental token on any enemy
+	  before resolving that spell's effect.
+      </p>
+      `,
+    keywords: ['develop', 'elementaltoken'],
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/cards.ts:52]: Add 'attach' keyword.

File: src/aer-data/src/ENG/theCaverns/cards.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    id: 'CombustLegend',
+    cost: 4,
+    effect: `
+      <p>
+      While prepped, after any player plays
+	  a relic, place an <span class="aether">&AElig;</span> token on this.<br/>
+	  <b>Cast:</b> Deal 3 damage. Gain all of the <span class="aether">&AElig;</span> tokens on this.
+      </p>
+      `,
+    keywords: [''],
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/cards.ts:82]: Delete empty string keyword.

File: src/aer-data/src/ENG/theCaverns/cards.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    effect: `
+      <p>
+      While prepped, after you play a
+	  relic, place a charge token on this.<br/>
+	  <b>Cast:</b> Deal 4 damage. Deal an additional 2
+	  damage for each charge token on this. Then,
+	  discard all of the tokens placed on this.
+      </p>
+      `,
+    keywords: [''],
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/cards.ts:99]: Delete empty string keyword.

File: src/aer-data/src/ENG/theCaverns/cards.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    cost: 8,
+    effect: `
+      <p>
+	  <b>Cast:</b> Deal 6 damage. Set aside any
+	  number of cards in hand. After drawing
+	  cards during your draw phase, return
+	  the set-aside cards to your hand.
+      </p>
+      `,
+    keywords: [''],
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/cards.ts:115]: Delete empty string keyword.

**Review by Torgen — CHANGES_REQUESTED** [review]

**Comment by deusabianca** [comment]
Thanks @Torgen - FYI I also changed 'elementaltoken' to 'elemental token'

**Comment by Torgen** [comment]
FYI, am pretty close to having the Descent data entered, including friends and foes. Will probably make that a single PR, then a separate one for the code changes to make the friends and foes selectable in settings and allow picking them at random.

File: src/aer-data/src/ENG/talesOfOldGravehold/nemeses.ts
Comment (Torgen) [inline src/aer-data/src/ENG/talesOfOldGravehold/nemeses.ts:1]: Is this nemesis in both expansions?

Comment (deusabianca) [reply src/aer-data/src/ENG/talesOfOldGravehold/nemeses.ts:1]: Whoops, I meant to make this a nemesis file without any nemeses as the ruins and buried secrets folder has and see if that fixed the tests.

**Comment by deusabianca** [comment]
> FYI, am pretty close to having the Descent data entered, including friends and foes.

Awesome ty @Torgen. I can work on the promos and The Abyss, though you could add the friends and foes from The Caverns and The Abyss?

**Comment by Torgen** [comment]
Option 1 is if my PR gets merged first, you rebase so you can see the type changes that add friends and foes, then I do a PR into your branch to add them. Option 2 is you clone this PR into one against my branch and we end up with one big PR.

**Comment by deusabianca** [comment]
All content from the minis aside from friends and foes and banners should be in this PR now.

@Torgen option 1 sounds good

**Comment by Torgen** [comment]
There are no friends, foes, or banners in TOG, right? We didn't get that one.

**Comment by deusabianca** [comment]
TOG only has 2 supply cards and a mage

File: src/aer-data/src/ENG/theCaverns/foes.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        `,
+      },
+    ],
+    extraCards: [
+      {
+        name: 'Shock',
+        id: 'Shock',
+        expansion: 'TC',
+        type: 'Spell',
+        cost: 0
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/foes.ts]: Needs comma.

File: src/aer-data/src/ENG/theCaverns/foes.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+          The Storm gains 2 charges.
+          <span class="or">OR</span>
+          Any player discards a prepped
+		  spell, gains a Shock, and places
+		  it on top of their deck.
+          </p>
+        `,
+      },
+      {
+        name: 'Fratctalize',
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/foes.ts:41]: Has an extra t.

File: src/aer-data/src/ENG/theCaverns/friends.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
+import { Friend } from '../../../../aer-types/types'
+
+export const friends: Friend[] = [
+  {
+    name: 'Joanna, the Merchant',
+    id: 'JoannatheMerchant',
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/friends.ts:6]: Other friends have a capital T on their The in the id.

File: src/aer-data/src/ENG/theCaverns/friends.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      },
+      {
+        name: 'Salvaged Baubles',
+        id: 'SalvagedBaubles',
+        expansion: 'TC',
+        type: 'Attack',
+        effect: `
+          <p>
+          Joanna, the Merchant gains 1 charge.
+
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/friends.ts:72]: These are HTML. You need at least a `<br/>` to force these onto separate lines; a `</p><p>` will put a margin between them like it is on the card.

File: src/aer-data/src/ENG/theCaverns/banners.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    id: 'BannerOfOvercharge',
+    type: 'Spell',
+    cost: 0,
+    expansion: 'TC',
+    keywords: [],
+    effect: `
+      <p>
+      While prepped, once per turn,
+	  you may discard a card in hand. If
+	  you do, the friend gains 1 charge.
```
Comment (Torgen) [inline src/aer-data/src/ENG/theCaverns/banners.ts:15]: Needs a `<br/>` to put the Cast on a new line.

**Review by Torgen — CHANGES_REQUESTED** [review]

File: src/aer-data/src/ENG/TDPromos/cards.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
+import { ICard } from '../../../../aer-types/types'
+
+export const cards: ICard[] = [
+  {
+    type: 'Gem',
+    expansion: 'TDPromo',
+    name: 'Glutton\'s Jewel',
+    id: 'GluttonsJewel',
+    cost: 4,
```
Comment (Torgen) [inline src/aer-data/src/ENG/TDPromos/cards.ts:9]: Costs 6, doesn't develop.

File: src/aer-data/src/ENG/theAbyss/cards.ts [hunk 12]
```diff
@@ -0,0 +1,117 @@
+import { ICard } from '../../../../aer-types/types'
+
+export const cards: ICard[] = [
+  {
+    type: 'Gem',
+    expansion: 'AB',
+    name: 'Spirit Jewel',
+    id: 'SpiritJewel',
+    cost: 4,
+	developCost: 2,
```
Comment (Torgen) [inline src/aer-data/src/ENG/theAbyss/cards.ts:10]: Doesn't develop.

**Comment by on3iro** [comment]
Hey @deusabianca is this ready for a final review and merge?

**Comment by deusabianca** [comment]
@on3iro Yes should be good to go!

**Review by on3iro — DISMISSED** [review]

**Comment by on3iro** [comment]
@deusabianca thanks for the PR. Could you please do the following:

1. Squash all commits into a single one
2. Make sure to rebase the the current main branch into your branch and resolve all conflicts
3. Force push onto this branch again, to update the pr$

(I would've done this myself, however it seems you are probably using windows and your line endings are not configured properly. I tried to fix this quickly on my local machine, but didn't quite get it right and I am really short on time right now. I thought our `.gitattributes` file should've handled this already, but it seems that it's not yet configured properly....)

**Comment by deusabianca** [comment]
I am still very much a noob with Github so the best way I found to do that I made a new pull request. I hope that's fine and we can close this one.

**Comment by on3iro** [comment]
Sure no - problem :) thanks @deusabianca

**Pull request closed without merging**